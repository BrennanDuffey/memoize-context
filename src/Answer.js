import React, { Component } from 'react';

class Answer extends Component {
  constructor(props) {
    super(props);
    this.state = {

    }
  }

  render() {
    return (
      <p onClick={
        () => {
          this.props.toggleShowAnswer();
          this.props.updateIsCorrect(this.props.correctAnswer === this.props.answer);
        }
      }>{this.props.answer}</p>
    )  
  }
}

export default Answer;