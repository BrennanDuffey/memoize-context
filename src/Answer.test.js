import React from 'react';
import Answer from './Answer.js';
import { shallow } from 'enzyme';

let mockAnswer = 'Undefined.';
let mockCorrectAnswer = 'Undefined.';
let mockUpdateIsCorrect = jest.fn();
let mockToggleShowAnswer = jest.fn();
let mockUpdateCorrectAnswers = jest.fn();

describe('Answer components', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow (
      <Answer  
        answer={mockAnswer}
        correctAnswer={mockCorrectAnswer}
        updateIsCorrect={mockUpdateIsCorrect}
        toggleShowAnswer={mockToggleShowAnswer}
        updateCorrectAnswers={mockUpdateCorrectAnswers}
      />
    );
  });

  it('Should render correctly with all the information passed', () => {
      expect(wrapper).toMatchSnapshot();
  });

  it('Should call invoke handleClick when button is clicked', () => {
    wrapper.find('.question').simulate('click', () => {})
    expect(mockToggleShowAnswer).toBeCalled();
    expect(mockUpdateIsCorrect).toBeCalled()
  });

  it('Should call invoke handleClick when button is clicked', () => {
    wrapper.find('.question').simulate('keyDown', (e) => {keyCode: 13})
    expect(mockToggleShowAnswer).toBeCalled();
    expect(mockUpdateIsCorrect).toBeCalled()
  });
});