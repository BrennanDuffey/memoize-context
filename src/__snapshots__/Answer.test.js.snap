// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer components Should render correctly with all the information passed 1`] = `
<p
  className="question"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  Undefined.
</p>
`;
